services/impulse_svc/grpc_service/service.py

```diff
@@ -15,14 +16,45 @@
 from grpc_service.models import Challenge, UserChallenge, User, Reward, Team
 
 from django.db.models import Q
-from django.db.utils import IntegrityError
 
 
-tz = pytz.timezone('Europe/Berlin')
+def is_uuid(uuid_input: str, version=4) -> bool:
+    try:
+        UUID(uuid_input, version=version)
+    except ValueError:
+        return False
+    return True
+
+
+def is_iso_datetime(datetime_input: str) -> bool:
+    try:
+        datetime.fromisoformat(datetime_input)
+    except ValueError:
+        return False
+    return True
```

Comment:
How about returning a tuple with the parsed value or None?

Reply:
Do you mean something like this?
```python
def get_uuid(uuid_input: str, version=4) -> UUID | None:
  pass

def get_iso_datetime(datetime_input: str) -> datetime | None:
  pass
```